gtfs-realtime/spec/en/reference.md

```diff
@@ -340,9 +354,89 @@ Identification information for the vehicle performing the trip.
 
 | _**Field Name**_ | _**Type**_ | _**Cardinality**_ | _**Description**_ |
 |------------------|------------|-------------------|-------------------|
-| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field |
+| **id** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | Internal system identification of the vehicle. Should be **unique** per vehicle, and is used for tracking the vehicle as it proceeds through the system. This id should not be made visible to the end-user; for that purpose use the **label** field. |
 | **label** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | User visible label, i.e., something that must be shown to the passenger to help identify the correct vehicle. |
 | **license_plate** | [string](https://developers.google.com/protocol-buffers/docs/proto#scalar) | optional | The license plate of the vehicle. |
+| **carriage_descriptor** | [CarriageDescriptor](#message-carriagedescriptor) | repeated | An ordered list of carriage information. |
```

Comment:
@barbeau Well, here in Europe it is not that rare for a train to change its direction (even multiple times) and still hold its designation, i.e. being modeled as a single trip. At least this is how I handle it. It'd be great if @RachM could check how it is modeled for trains running through e.g. Leipzig Hbf (DB on route Berlin-Erfurt), Hamburg Hbf (direct ICE trains on Berlin-Copenhagen route), Munich Hbf (OEBB on route Salzburg-Ulm) or Kristianstad (Öresundståg on Copenhagen-Karlskrona route). All major stations in Budapest are dead-ends as well, so long distance trains have to change direction there too.

I'm afraid employing direction_id for that purpose doesn't feel right to me semantically. If we would take this way I'd suggest a separate field for it, something like `trip/consist_orientation=FORWARD|BACKWARD`.

But I'm afraid there is another drawback with this approach as there could be only a single `TripDescriptor` (thus single `CarriageDescriptor[]`) at a time. So up till the time of arrival to the dead-end station the `CarriageDescriptor[]` would be reversed for the rest of the journey after such station. I think it'd be better to use approach similar to headsigns - i.e. `trip_headsign` can be overridden with `stop_headsing` later on. Such approach would enable us to describe even adding and removing carriages along the route of the train as well (again a technique still used for long distance trains in Europe). I know it complicates things, but I'm afraid that's reality :). Thoughts?